File: Java/Linked List/Singly Circular Linked List/SinglyCL09.java
/*
=====================================================================

Name            : Singly Circular Linked List - Constructor Initialization
Program Number  : 009
Description     : This program demonstrates the creation of a
                  Singly Circular Linked List class with a
                  parameterized node constructor and a list
                  constructor.

                  The constructor initializes the first pointer,
                  last pointer, and node count to their default
                  values.

Concepts Used   : Singly Circular Linked List,
                  Constructor,
                  Parameterized Constructor,
                  Object-Oriented Programming,
                  Reference Variables,
                  Class Initialization

Input           : None

Output          : Object creation of Singly Circular Linked List

Date            : July 2026
Language        : Java
Author          : Atharv Tushar Bhosale

=====================================================================
*/

class node
{
    public int data;
    public node next;

    node(int no)
    {
        this.data = no;
        this.next = null;
    }
}

class SinglyCL
{
    public node first;
    public node last;
    public int iCount;

    public SinglyCL()
    {
        this.first = null;
        this.last = null;
        this.iCount = 0;
    }
}

class SinglyCL09
{
    public static void main(String A[])
    {
        SinglyCL sobj = new SinglyCL();
    }
}

/*
=====================================================================

Sample Output :

(No Output)

=====================================================================

Observation :

- Node class contains a parameterized constructor.
- SinglyCL class contains first and last references.
- Constructor initializes list to empty state.
- Node count is initialized to zero.
- Linked List object is created successfully.

=====================================================================
*/

File: C++/Linked List/Singly Circular Linked List/SinglyCL09.cpp
/*
=====================================================================

Name            : Singly Circular Linked List - Function Prototypes
Program Number  : 009

Description     : This program declares all major operations
                  required for implementation of a Singly
                  Circular Linked List.

                  Function declarations are added for future
                  insertion, deletion, display and count
                  operations.

Concepts Used   : Classes,
                  Member Functions,
                  Function Prototypes,
                  Linked List Design

Input           : None

Output          : Program executed successfully

Date            : July 2026
Language        : C++
Author          : Atharv Tushar Bhosale

=====================================================================
*/

#include<iostream>
using namespace std;

class node
{
    public:
        int data;
        node* next;

        node(int no)
        {
            data = no;
            next = NULL;
        }
};

class SinglyCL
{
    public:
        node* first;
        node* last;
        int iCount;

        SinglyCL()
        {
            first = NULL;
            last = NULL;
            iCount = 0;
        }

        void Display()
        {}

        int Count()
        {
            return iCount;
        }

        void InsertFirst(int no)
        {}

        void InsertLast(int no)
        {}

        void InsertAtPos(int no,int pos)
        {}

        void DeleteFirst()
        {}

        void DeleteLast()
        {}

        void DeleteAtPos(int pos)
        {}
};

int main()
{
    SinglyCL sobj;

    return 0;
}

/*
=====================================================================

Output :

Program executed successfully

=====================================================================
*/
